src/app/server/Server.h

```diff
@@ -440,10 +426,23 @@ class Server
         void OnFabricNOCUpdated(chip::FabricTable & fabricTable, chip::FabricIndex fabricIndex) override
         {
             (void) fabricTable;
-            (void) fabricIndex;
+            ClearCASEResumptionStateOnFabricChange(fabricIndex);
         }
 
     private:
+        void ClearCASEResumptionStateOnFabricChange(chip::FabricIndex fabricIndex)
+        {
+            auto * sessionResumptionStorage = mServer->GetSessionResumptionStorage();
+            VerifyOrReturn(sessionResumptionStorage != nullptr);
+            CHIP_ERROR err = sessionResumptionStorage->DeleteAll(fabricIndex);
```

Comment:
As a followup: we need this cleanup in controller too, right?

Reply:
I don't think so. That already calls `ReturnErrorOnFailure(params.systemState->Fabrics()->UpdateFabric(mFabricInfo->GetFabricIndex(), newFabric));` which would end up calling this right?

Reply:
It won't end up calling this, because a pure controller is not running a `Server` instance at all.  That's why controller has a separate delegate for this stuff that duplicates this one.

Reply:
:man_facepalming: thanks for explaining that I will do that for controller too